langchain4j-azure-cosmos/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/mongo/vcore/AzureCosmosDbMongoVCoreEmbeddingStore.java

```diff
@@ -0,0 +1,622 @@
+         *                 of 1 is akin to performing brute-force search, which has limited performance.
+         * @return
+         */
+        public Builder numLists(Integer numLists) {
+            this.numLists = numLists;
+            return this;
+        }
+
+        /**
+         * @param similarity - Similarity metric to use with the index.
+         *                   Possible options are:
+         *                   - COS (cosine distance),
+         *                   - L2 (Euclidean distance), and
+         *                   - IP (inner product).
+         * @return
+         */
+        public Builder similarity(String similarity) {
+            this.similarity = similarity;
+            return this;
+        }
+
+        /**
+         * @param dimensions - Number of dimensions for vector similarity. The maximum number of supported dimensions
+         *                   is 2000.
+         * @return
+         */
+        public Builder dimensions(Integer dimensions) {
+            this.dimensions = dimensions;
+            return this;
+        }
+
+        /**
+         * @param m - The max number of connections per layer (16 by default, minimum value is 2, maximum
+         *          value is 100). Higher m is suitable for datasets with high dimensionality and/or high
+         *          accuracy requirements.
+         * @return
+         */
+        public Builder m(Integer m) {
+            this.m = m;
+            return this;
+        }
+
+        /**
+         * @param efConstruction - the size of the dynamic candidate list for constructing the graph (64 by default, minimum
+         *                       value is 4, maximum value is 1000). Higher ef_construction will result in better index
+         *                       quality and higher accuracy, but it will also increase the time required to build the index.
+         *                       ef_construction has to be at least 2 * m.
+         * @return
+         */
+        public Builder efConstruction(Integer efConstruction) {
+            this.efConstruction = efConstruction;
+            return this;
+        }
+
+        /**
+         * @param efSearch - The size of the dynamic candidate list for search (40 by default). A higher value provides
+         *                 better recall at the cost of speed.
+         * @return
+         */
+        public Builder efSearch(Integer efSearch) {
+            this.efSearch = efSearch;
+            return this;
+        }
+
+        public AzureCosmosDbMongoVCoreEmbeddingStore build() {
+            return new AzureCosmosDbMongoVCoreEmbeddingStore(mongoClient, connectionString, databaseName, collectionName, indexName, applicationName,
+                    createCollectionOptions, createIndex, kind, numLists, similarity, dimensions, m,
+                    efConstruction, efSearch);
+        }
+    }
+
+    public enum AzureCosmosDBSimilarityType {
```

Comment:
```suggestion
    public enum SimilarityMetric {
```

No need to duplicate "AzureCosmosDB" if it is an inner class

Reply:
Updated the name